File: api-tester/src/main/java/com/revenuecat/apitester/java/GoogleProrationModeAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.models.GoogleProrationMode;

@SuppressWarnings({"unused", "SpellCheckingInspection"})
final class GoogleProrationModeAPI {
    static void check(final GoogleProrationMode mode) {
        switch (mode) {
            case IMMEDIATE_WITHOUT_PRORATION:
            case IMMEDIATE_WITH_TIME_PRORATION:
            case DEFERRED:
            case IMMEDIATE_AND_CHARGE_FULL_PRICE:
            case IMMEDIATE_AND_CHARGE_PRORATED_PRICE:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/GoogleProrationModeAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.models.GoogleProrationMode

@Suppress("unused", "UNUSED_VARIABLE")
private class GoogleProrationModeAPI {
    fun check(mode: GoogleProrationMode) {
        when (mode) {
            GoogleProrationMode.IMMEDIATE_WITHOUT_PRORATION,
            GoogleProrationMode.IMMEDIATE_WITH_TIME_PRORATION,
            GoogleProrationMode.DEFERRED,
            GoogleProrationMode.IMMEDIATE_AND_CHARGE_FULL_PRICE,
            GoogleProrationMode.IMMEDIATE_AND_CHARGE_PRORATED_PRICE -> {}
        }.exhaustive
    }
}
